src/components/Form/Checkbox.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { Component, ReactNode } from 'react';
+import classnames from 'classnames';
+import { HtmlInput, HtmlLabel, HtmlSpan, HtmlDiv } from '../../reset';
+
+import { idGenerator } from '../../utils/uuid';
+
+const baseClassName = 'fi-checkbox';
+
+export const checkboxBaseClassNames = {
+  disabled: `${baseClassName}--disabled`,
+  container: `${baseClassName}_container`,
+  input: `${baseClassName}_input`,
+  label: `${baseClassName}_label`,
+  statusText: `${baseClassName}_status`,
+  hintText: `${baseClassName}_hintText`,
+};
```

Comment:
Even better, create a getter that always includes the baseClassName?

Reply:
import { getClassName } from 'components/checkbox';

getClassName().disabled; vs. checkboxBaseClassNames.disabled?

Maybe the latter is better. And I propose to change the object name to checkboxClassName, since the base refers to prefix fi-checkbox.

Reply:
And now that I read this PR more thoroughly, it's also a bit confusing to have checkboxBaseClassNames here with the --disabled style, then we have error and other derived styles elsewhere and the relationship between components and core styles and classnames is difficult to understand in my opinion. If these have an effect on the core styles as well or if core styles are built on top of these, I think everything that is put together in core should make sense / be consistent and clear in a way that it's easy to understand what comes from components, what from core and that there are no conflicts. Having e.g. baseClassName with the composed object here and then using baseClassName = "fi-checkbox..." in core is a bit confusing.

Reply:
We have a lot of these elsewhere, but I think this could be improved.

Reply:
I noticed an issue with some of the classnames. We indicate status with some of the classnames which is ok. However, in some cases we apply the status indicating class to underlying elements, like svg in toggle and the svg does not have a reliable selector to use with testing. These classnames do indicate a status as well. I guess it would be ok to leave these as they are now to be consistent with older components. Is there a reason to have some of the classnames / styles in component files and other here?

Reply:
Classes other than base classes should now be in one place (core).